### PR TITLE
Overhaul Travis & Codecov setup

### PR DESCRIPTION
**Pull request by fingolfin**
For Codecov, actually enable status messages in pull requests (so far,
all were turned off, but for some reason codecov ignores this; still,
better to specify actual intent in the file).

For Travis, adapt scripts/build_pkg.sh to work with packages not using
autoconf, so that we can use the same script across many packages. Also
don't copy the tested package into GAP's pkg dir; instead, put it into
its own GAP root which we tell GAP to look at before its usual GAP root.

Modify scripts/gather-coverage.sh and scripts/run_tests.sh to use this
modified list of GAP root dirs

Simplify scripts/build_gap.sh significantly by using BuildPackages.sh.

### CODE REVIEW

**Comment by fingolfin** [comment]
Please do not merge this just yet: the profiling compilation failed (due to `clang++` issue), which is a problem in itself, but also the fact that this did not cause the build to be marked as failed.

Of course it's a questionable why this package is tested with clang in the first place, as it contains no compiled code?

**Comment by fingolfin** [comment]
This is now waiting on https://github.com/gap-system/gap/pull/2022

**Comment by olexandr-konovalov** [comment]
> Of course it's a questionable why this package is tested with clang in the first place, as it contains no compiled code?

Setup was inherited from some other package, which used compiled code. It makes no harm to test a package without compiled code on different setting as well, isn't it? For example, FactInt, I assume, may exercise GMP heavily.

**Comment by fingolfin** [comment]
I don't think it exercises GMP more heavily than the GAP test suite. It mostly performs elementary operations only.

Anyway, this PR is now IMHO ready for merge.

**Comment by olexandr-konovalov** [comment]
OK - I want to edit the history a bit, should that be done now, or after?

**Comment by fingolfin** [comment]
Which history? That of the master branch of this package? Or of this PR? I am fine either way, just if you do it before merging the PR, let me know, because I'll have to rebase it.

That said, actually I'd like to edit this PR *some more* to make it trivial to add tests against the stable branch and/or HPC-GAP in the future (if desired). So, just go ahead with editing the `master` history :-)

**Comment by olexandr-konovalov** [comment]
@fingolfin I want to edit master history to make changes within last two weeks atomic. I will ask you and @Stefan-Kohl to reset your repositories then.

**Comment by Stefan-Kohl** [comment]
Should I merge this?

**Comment by olexandr-konovalov** [comment]
@fingolfin not yet, please- I want first to understand why Example package fails, with the same setup.

**Review by olexandr-konovalov — CHANGES_REQUESTED** [review]
Please rebase after resetting the master branch of your clone/fork.

**Comment by fingolfin** [comment]
Rebased

**Review by olexandr-konovalov — APPROVED** [review]

**Pull request merged**
